### PR TITLE
don't notify users who've told us they can't vote

### PR DESCRIPTION
**Pull request by theJohnnyBrown**

### CODE REVIEW

File: src/election_notification_works/registration_status.clj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+
+(defn not-ineligible?
+  "This is used to pre-empt notifications, so we return true unles a user has
+  positively identified as ineligible"
+  [user-id]
+  (let [resp (clojure.core.async/<!! (reg-status-read [user-id]))]
+    (case (:status resp)
+      :ok (not= (-> resp :registration-status :status) :ineligible)
+      :not-found true
```
Comment (eggsyntax) [inline src/election_notification_works/registration_status.clj]: In what sorts of cases would `:status` be `:not-found`?

File: test/election_notification_works/handlers/signup_test.clj [hunk 2]
```diff
@@ -226,100 +232,61 @@
...
-              expected [signup-result confirm-result]
-              results (atom [])]
-          (reschedule (success-search-fn (list {:message signup-message}
-                                               {:message confirm-message}))
-                      (fn [m] (swap! results conj m) (async/go {:status :ok}))
-                      user-id)
-          (is (= expected @results))))))
-
-  ;; The following tests check that signup events trigger a reschedule
-  (letfn [(test-event [event-fn]
```
Comment (eggsyntax) [inline test/election_notification_works/handlers/signup_test.clj]: Out of curiosity, why get rid of these tests? Would they somehow no longer be relevant? I feel like I'm missing something.

File: src/election_notification_works/registration_status.clj (hunk 1 above)
Comment (theJohnnyBrown) [reply src/election_notification_works/registration_status.clj]: In cases where `voter-registration-works` has no record for the given user ID. For example, users who haven't reached that step of their signup flow will be `not-found`.

File: test/election_notification_works/handlers/signup_test.clj (hunk 2 above)
Comment (theJohnnyBrown) [reply test/election_notification_works/handlers/signup_test.clj]: No, I didn't mean to do that. It should be just indentation changes from wrapping things in `with-redefs`. I'll take a closer look.

Comment (eggsyntax) [reply test/election_notification_works/handlers/signup_test.clj]: Oh, could definitely be github messing up the diff. I told it to ignore whitespace, and I've certainly seen that go wrong occasionally...

File: src/election_notification_works/registration_status.clj (hunk 1 above)
Comment (eggsyntax) [reply src/election_notification_works/registration_status.clj]: Makes sense, thanks!

**Comment by katrinamariehh** [comment]
💅🏻lgtm

**Pull request merged**